activesupport/test/core_ext/hash_ext_test.rb

```diff
@@ -1523,9 +1523,9 @@ def test_should_use_default_proc_for_unknown_key
     assert_equal 3, hash_wia[:new_key]
   end
 
-  def test_should_use_default_proc_if_no_key_is_supplied
+  def test_should_return_nil_if_no_key_is_supplied
     hash_wia = HashWithIndifferentAccess.new { 1 +  2 }
-    assert_equal 3, hash_wia.default
+    assert_equal nil, hash_wia.default
```

Comment:
confirm. https://github.com/rails/rails/pull/16356/files#diff-c5e8b13afcecaf67badb95d97f9e860cR1526 is wrong: